tools/hermes-parser/js/hermes-estree/src/types.js

```diff
@@ -1560,7 +1560,7 @@ export interface InterfaceDeclaration extends BaseInterfaceDeclaration {
 
 export interface InterfaceExtends extends BaseNode {
   +type: 'InterfaceExtends';
-  +id: Identifier | QualifiedTypeIdentifier;
+  +id: Identifier | QualifiedTypeIdentifier | MemberExpression | TypeAnnotationType;
```

Comment:
This type change is incorrect: https://astexplorer.net/#/gist/5c8ba16b70de7482e04a1e4bc9d481fb/3d30f92fa471d423d4569497a778a73eabd8f042

Both `MemberExpression` and `TypeAnnotationType` are invalid in these locations (although they may happened to print correct atm). You need to instead convert these back to the appropriate supported AST types.

Reply:
Thanks for the review. I'm working with @huntie to improve the coverage in [public-api-test](https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/__tests__/public-api-test.js).

Are there existing functions that do this conversion ? I see some similar logic in `TSDefToFlowDef`, but so far nothing that "just works". If not, I can try adding them -- just want to verify before committing time.